src/util/Database.js

```diff
@@ -13,12 +13,11 @@ export default class Database {
   }
 
   static addResearchData(researcherUserId, researcherData) {
-    firebase
+    return firebase
```

Comment:
Unlike the functions in `Authentication.js`, since we are not parsing the errors here and re-throwing them, if we catch the errors here, the Promise will be resolved here; thus, nothing is returned and errors will not be propagated to `SignUpResearcher.js`. Instead, we return this promise here, and await for this promise inside the try catch block in `SignUpResearcher.js` such that if there's any error, it will be caught and the error notification displaying code in the catch block can be executed.